modules/end-user-guide/partials/proc_building-default-intellij-editor-images.adoc

```diff
@@ -0,0 +1,91 @@
+[id="building-default-intellij-editor-images_{context}"]
+= Building default IntelliJ editor images
+
+This procedure describes how to build an IntelliJ-based editor image using the default values proposed by the build script.
+
+.Prerequisites
+
+* Define the following locations to publish the editor image and metadata:
++
+`__<IntelliJ-container-image>__`::
+Location of the IntelliJ editor image in a registry accessible by {prod-short}.
++
+`__<URL_to_meta.yaml>__`::
+HTTPS resource containing the editor metadata.
+
+* A Docker version supporting link:https://docs.docker.com/develop/develop-images/build_enhancements/[Docker BuildKit]. Docker 18.09 or greater.
+* The `git` and `jq`tools are available.
+* On the macOS platform: the `+gnu-getopt+` tool is available.
+
+.Procedure
+
+. Get a local copy of the link:https://github.com/che-incubator/jetbrains-editor-images[JetBrains Projector Editor Images repository].
++
+----
+$ git clone https://github.com/che-incubator/jetbrains-editor-images
+$ cd jetbrains-editor-images
+----
+
+. Build JetBrains IDE based on the predefined configuration, and reply to the prompt to identify the IDE name and version to build:
++
+----
+$ ./projector.sh build
```

Comment:
Now I added wget to the prerequisites and I still have this build error. I guess there are some more missing prerequistes. Any idea @vzhukovs ?

```
vl stop  fedora-34 && vl up && vl ansible_inventory > inventory && ansible-playbook -i inventory playbook.yml && vl ssh fedora-34
⚡ fedora-34 
⌛ ok Waiting...
🛃 fedora-34 QEMU agent found
💻 fedora-34 found at 192.168.123.5!
👍 You are all set

PLAY [prerequisites and partial procedure] ******************************************************************************************************

TASK [prerequisites - docker - dnf-plugins-core] ************************************************************************************************
Friday 03 September 2021  11:01:12 +0200 (0:00:00.052)       0:00:00.052 ****** 
Friday 03 September 2021  11:01:12 +0200 (0:00:00.050)       0:00:00.050 ****** 
changed: [fedora-34]

TASK [prerequisites - docker - repository] ******************************************************************************************************
Friday 03 September 2021  11:02:26 +0200 (0:01:14.146)       0:01:14.199 ****** 
Friday 03 September 2021  11:02:26 +0200 (0:01:14.146)       0:01:14.197 ****** 
changed: [fedora-34]

TASK [prerequisites - docker - packages] ********************************************************************************************************
Friday 03 September 2021  11:02:27 +0200 (0:00:00.905)       0:01:15.104 ****** 
Friday 03 September 2021  11:02:27 +0200 (0:00:00.905)       0:01:15.102 ****** 
changed: [fedora-34]

TASK [prerequisites - docker - service] *********************************************************************************************************
Friday 03 September 2021  11:03:08 +0200 (0:00:41.562)       0:01:56.666 ****** 
Friday 03 September 2021  11:03:08 +0200 (0:00:41.562)       0:01:56.664 ****** 
changed: [fedora-34]

TASK [prerequisites - packages - git jq java wget] ***************************************************************************************************
Friday 03 September 2021  11:03:11 +0200 (0:00:02.124)       0:01:58.790 ****** 
Friday 03 September 2021  11:03:11 +0200 (0:00:02.124)       0:01:58.788 ****** 
changed: [fedora-34]

TASK [procedure - clone the repository] *********************************************************************************************************
Friday 03 September 2021  11:03:47 +0200 (0:00:36.224)       0:02:35.015 ****** 
Friday 03 September 2021  11:03:47 +0200 (0:00:36.224)       0:02:35.013 ****** 
changed: [fedora-34]

TASK [procedure - interactive steps] ************************************************************************************************************
Friday 03 September 2021  11:03:49 +0200 (0:00:02.241)       0:02:37.257 ****** 
Friday 03 September 2021  11:03:49 +0200 (0:00:02.241)       0:02:37.255 ****** 
ok: [fedora-34] => {
    "msg": "Execute following interactive steps: . Connect to the virtual machine using SSH $ vl ssh fedora-34 . In the SSH session, build the project: # cd jetbrains-editor-images && ./projector.sh build\n"
}

PLAY RECAP **************************************************************************************************************************************
fedora-34                  : ok=7    changed=6    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

Friday 03 September 2021  11:03:49 +0200 (0:00:00.026)       0:02:37.284 ****** 
=============================================================================== 
ansible.builtin.dnf --------------------------------------------------- 151.93s
ansible.builtin.git ----------------------------------------------------- 2.24s
ansible.builtin.systemd ------------------------------------------------- 2.12s
ansible.builtin.yum_repository ------------------------------------------ 0.91s
debug ------------------------------------------------------------------- 0.03s
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
total ----------------------------------------------------------------- 157.23s
Friday 03 September 2021  11:03:49 +0200 (0:00:00.027)       0:02:37.282 ****** 
=============================================================================== 
prerequisites - docker - dnf-plugins-core ----------------------------------------------------------------------------------------------- 74.15s
prerequisites - docker - packages ------------------------------------------------------------------------------------------------------- 41.56s
prerequisites - packages - git jq java -------------------------------------------------------------------------------------------------- 36.22s
procedure - clone the repository --------------------------------------------------------------------------------------------------------- 2.24s
prerequisites - docker - service --------------------------------------------------------------------------------------------------------- 2.12s
prerequisites - docker - repository ------------------------------------------------------------------------------------------------------ 0.91s
procedure - interactive steps ------------------------------------------------------------------------------------------------------------ 0.03s
Playbook run took 0 days, 0 hours, 2 minutes, 37 seconds
Warning: Permanently added '192.168.123.5' (ED25519) to the list of known hosts.
Last login: Fri Sep  3 09:03:47 2021 from 192.168.123.1
[ffloreth@fedora-34 ~]$ cd jetbrains-editor-images
[ffloreth@fedora-34 jetbrains-editor-images]$ ./projector.sh build
[info] Select the IDE package to build (default is 'IntelliJ IDEA Community'):
       1) IntelliJ IDEA Community
       2) PyCharm Community
#? 1
[info] Select the IDE package version to build (default is '2020.3.3'):
       1) 2020.3.3
       2) 2020.3.2
       3) 2020.3.1
#? 1
[info] Read the container name 'che-idea:2020.3.3'
[info] Read the URL for IDE packaging 'https://download-cdn.jetbrains.com/idea/ideaIC-2020.3.3.tar.gz'
[info] Check for installed Java
[info] Read JAVA_HOME env: 
[info] Read PATH env: /home/ffloreth/.local/bin:/home/ffloreth/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin
[info] Found 'java' executable in PATH
[info] Java version: 16.0.2
--2021-09-03 09:07:51--  https://download-cdn.jetbrains.com/idea/ideaIC-2020.3.3.tar.gz
Resolving download-cdn.jetbrains.com (download-cdn.jetbrains.com)... 13.225.19.37, 13.225.19.94, 13.225.19.43, ...
Connecting to download-cdn.jetbrains.com (download-cdn.jetbrains.com)|13.225.19.37|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 734643423 (701M) [binary/octet-stream]
Saving to: ‘ideaIC-2020.3.3.tar.gz’

ideaIC-2020.3.3.tar.gz               100%[===================================================================>] 700.61M  8.97MB/s    in 67s     

2021-09-03 09:08:59 (10.4 MB/s) - ‘ideaIC-2020.3.3.tar.gz’ saved [734643423/734643423]

[info] Check Projector Client source directory
[warning] Projector Client source directory '/home/ffloreth/jetbrains-editor-images/build/projector-client' doesn't exist
[info] Cloning Projector Client sources to '/home/ffloreth/jetbrains-editor-images/build/projector-client'
[info] Checkout Projector Client to SHA1 'd80f9952cc520102bb2de6f1aefaad6aadb49138'
[info] HEAD is now at d80f995 PRJ-541 PRJ-542 Show CloseEvent details on disconnection
[info] Applying patches for Projector Client
       Patching Projector Client with '/home/ffloreth/jetbrains-editor-images/patches/projector-client/PRJ-452.patch'
[info] Check Projector Server source directory
[warning] Projector Server source directory '/home/ffloreth/jetbrains-editor-images/build/projector-server' doesn't exist
[info] Cloning Projector Server sources to '/home/ffloreth/jetbrains-editor-images/build/projector-server'
[info] Checkout Projector Client to SHA1 'e8ecfda9ee04b59e8b4276215e22a3cbff212083'
[info] HEAD is now at e8ecfda New release
[info] Build Projector on localhost
The message received from the daemon indicates that the daemon has disappeared.
Build request sent: Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server}
Attempting to read last messages from the daemon log...
Daemon pid: 11250
  log file: /home/ffloreth/.gradle/daemon/6.4.1/daemon-11250.out.log
----- Last  20 lines from daemon log file - daemon-11250.out.log -----
2021-09-03T09:09:20.700+0000 [INFO] [org.gradle.launcher.daemon.server.DefaultIncomingConnectionHandler] Received command: Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server}.
2021-09-03T09:09:20.700+0000 [DEBUG] [org.gradle.launcher.daemon.server.DefaultIncomingConnectionHandler] Starting executing command: Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server} with connection: socket connection from /127.0.0.1:44471 to /127.0.0.1:57178.
2021-09-03T09:09:20.703+0000 [ERROR] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] Command execution: started DaemonCommandExecution[command = Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server}, connection = DefaultDaemonConnection: socket connection from /127.0.0.1:44471 to /127.0.0.1:57178] after 0.0 minutes of idle
2021-09-03T09:09:20.703+0000 [INFO] [org.gradle.launcher.daemon.server.DaemonRegistryUpdater] Marking the daemon as busy, address: [980bd925-1bbb-444d-8a01-e5f3f87f7f35 port:44471, addresses:[localhost/127.0.0.1]]
2021-09-03T09:09:20.703+0000 [DEBUG] [org.gradle.launcher.daemon.registry.PersistentDaemonRegistry] Marking busy by address: [980bd925-1bbb-444d-8a01-e5f3f87f7f35 port:44471, addresses:[localhost/127.0.0.1]]
2021-09-03T09:09:20.706+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire exclusive lock on daemon addresses registry.
2021-09-03T09:09:20.706+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2021-09-03T09:09:20.708+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2021-09-03T09:09:20.709+0000 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] resetting idle timer
2021-09-03T09:09:20.709+0000 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] daemon is running. Sleeping until state changes.
2021-09-03T09:09:20.711+0000 [INFO] [org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy] Daemon is about to start building Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server}. Dispatching build started information...
2021-09-03T09:09:20.711+0000 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 21: dispatching org.gradle.launcher.daemon.protocol.BuildStarted@582ccbc9
2021-09-03T09:09:20.714+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment] Configuring env variables: {PATH=/home/ffloreth/.local/bin:/home/ffloreth/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin, LC_MEASUREMENT=C.UTF-8, LC_MONETARY=C.UTF-8, HISTCONTROL=ignoredups, XMODIFIERS=@im=ibus, LC_TIME=C.UTF-8, HISTSIZE=1000, BASH_FUNC_which%%=() {  ( alias;
 eval ${_declare} ) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot "$@"
}, TERM=xterm-256color, MOTD_SHOWN=pam, DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus, LANG=en_US.UTF-8, XDG_SESSION_ID=3, XDG_SESSION_TYPE=tty, SELINUX_LEVEL_REQUESTED=, SELINUX_ROLE_REQUESTED=, MAIL=/var/spool/mail/ffloreth, LC_PAPER=C.UTF-8, LOGNAME=ffloreth, PWD=/home/ffloreth/jetbrains-editor-images/build/projector-server, XDG_SESSION_CLASS=user, LESSOPEN=||/usr/bin/lesspipe.sh %s, SHELL=/bin/bash, SELINUX_USE_CURRENT_RANGE=, SSH_TTY=/dev/pts/0, SSH_CLIENT=192.168.123.1 41862 22, OLDPWD=/home/ffloreth/jetbrains-editor-images/build/projector-server, USER=ffloreth, LC_NUMERIC=C.UTF-8, SSH_CONNECTION=192.168.123.1 41862 192.168.123.5 22, HOSTNAME=fedora-34, XDG_RUNTIME_DIR=/run/user/1000, LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=01;37;41:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=01;36:*.au=01;36:*.flac=01;36:*.m4a=01;36:*.mid=01;36:*.midi=01;36:*.mka=01;36:*.mp3=01;36:*.mpc=01;36:*.ogg=01;36:*.ra=01;36:*.wav=01;36:*.oga=01;36:*.opus=01;36:*.spx=01;36:*.xspf=01;36:, SHLVL=2, HOME=/home/ffloreth}
2021-09-03T09:09:20.730+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.LogToClient] About to start relaying all logs to the client via the connection.
2021-09-03T09:09:20.730+0000 [INFO] [org.gradle.launcher.daemon.server.exec.LogToClient] The client will now receive all logging from the daemon (pid: 11250). The daemon log file: /home/ffloreth/.gradle/daemon/6.4.1/daemon-11250.out.log
2021-09-03T09:09:20.733+0000 [INFO] [org.gradle.launcher.daemon.server.exec.LogAndCheckHealth] Starting build in new daemon [memory: 519.0 MB]
2021-09-03T09:09:20.744+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] The daemon has started executing the build.
2021-09-03T09:09:20.747+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] Executing build with daemon context: DefaultDaemonContext[uid=89502f5f-bcdb-432e-9a05-6f26dcbe36ab,javaHome=/usr/lib/jvm/java-16-openjdk-16.0.2.0.7-1.rolling.fc34.x86_64,daemonRegistryDir=/home/ffloreth/.gradle/daemon,pid=11250,idleTimeout=10800000,priority=NORMAL,daemonOpts=--add-opens,java.base/java.util=ALL-UNNAMED,--add-opens,java.base/java.lang=ALL-UNNAMED,--add-opens,java.base/java.lang.invoke=ALL-UNNAMED,--add-opens,java.prefs/java.util.prefs=ALL-UNNAMED,-XX:MaxMetaspaceSize=256m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=UTF-8,-Duser.country=US,-Duser.language=en,-Duser.variant]
----- End of the daemon log -----


FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
find: ‘projector-server/build/distributions’: No such file or directory
[info] Build 'che-idea:2020.3.3'
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/build?buildargs=%7B%7D&buildid=36b96f3d31493cf73e0fd8a119343c73961690ef101e540bf58ae5e229f68365&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=default&remote=client-session&rm=1&session=ommuhhgz7j2d2iq9lpxa7mo8k&shmsize=0&t=che-idea%3A2020.3.3&target=&ulimits=null&version=2": dial unix /var/run/docker.sock: connect: permission denied
[warning] Container build failed
[ffloreth@fedora-34 jetbrains-editor-images]$ free
               total        used        free      shared  buff/cache   available
Mem:          734072      161372      445768        1356      126932      462012
Swap:              0           0           0
```

Reply:
> Now I added wget to the prerequisites and I still have this build error. I guess there are some more missing prerequistes. Any idea @vzhukovs ?
> 
> ```
> vl stop  fedora-34 && vl up && vl ansible_inventory > inventory && ansible-playbook -i inventory playbook.yml && vl ssh fedora-34
> ⚡ fedora-34 
> ⌛ ok Waiting...
> 🛃 fedora-34 QEMU agent found
> 💻 fedora-34 found at 192.168.123.5!
> 👍 You are all set
> 
> PLAY [prerequisites and partial procedure] ******************************************************************************************************
> 
> TASK [prerequisites - docker - dnf-plugins-core] ************************************************************************************************
> Friday 03 September 2021  11:01:12 +0200 (0:00:00.052)       0:00:00.052 ****** 
> Friday 03 September 2021  11:01:12 +0200 (0:00:00.050)       0:00:00.050 ****** 
> changed: [fedora-34]
> 
> TASK [prerequisites - docker - repository] ******************************************************************************************************
> Friday 03 September 2021  11:02:26 +0200 (0:01:14.146)       0:01:14.199 ****** 
> Friday 03 September 2021  11:02:26 +0200 (0:01:14.146)       0:01:14.197 ****** 
> changed: [fedora-34]
> 
> TASK [prerequisites - docker - packages] ********************************************************************************************************
> Friday 03 September 2021  11:02:27 +0200 (0:00:00.905)       0:01:15.104 ****** 
> Friday 03 September 2021  11:02:27 +0200 (0:00:00.905)       0:01:15.102 ****** 
> changed: [fedora-34]
> 
> TASK [prerequisites - docker - service] *********************************************************************************************************
> Friday 03 September 2021  11:03:08 +0200 (0:00:41.562)       0:01:56.666 ****** 
> Friday 03 September 2021  11:03:08 +0200 (0:00:41.562)       0:01:56.664 ****** 
> changed: [fedora-34]
> 
> TASK [prerequisites - packages - git jq java wget] ***************************************************************************************************
> Friday 03 September 2021  11:03:11 +0200 (0:00:02.124)       0:01:58.790 ****** 
> Friday 03 September 2021  11:03:11 +0200 (0:00:02.124)       0:01:58.788 ****** 
> changed: [fedora-34]
> 
> TASK [procedure - clone the repository] *********************************************************************************************************
> Friday 03 September 2021  11:03:47 +0200 (0:00:36.224)       0:02:35.015 ****** 
> Friday 03 September 2021  11:03:47 +0200 (0:00:36.224)       0:02:35.013 ****** 
> changed: [fedora-34]
> 
> TASK [procedure - interactive steps] ************************************************************************************************************
> Friday 03 September 2021  11:03:49 +0200 (0:00:02.241)       0:02:37.257 ****** 
> Friday 03 September 2021  11:03:49 +0200 (0:00:02.241)       0:02:37.255 ****** 
> ok: [fedora-34] => {
>     "msg": "Execute following interactive steps: . Connect to the virtual machine using SSH $ vl ssh fedora-34 . In the SSH session, build the project: # cd jetbrains-editor-images && ./projector.sh build\n"
> }
> 
> PLAY RECAP **************************************************************************************************************************************
> fedora-34                  : ok=7    changed=6    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
> 
> Friday 03 September 2021  11:03:49 +0200 (0:00:00.026)       0:02:37.284 ****** 
> =============================================================================== 
> ansible.builtin.dnf --------------------------------------------------- 151.93s
> ansible.builtin.git ----------------------------------------------------- 2.24s
> ansible.builtin.systemd ------------------------------------------------- 2.12s
> ansible.builtin.yum_repository ------------------------------------------ 0.91s
> debug ------------------------------------------------------------------- 0.03s
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
> total ----------------------------------------------------------------- 157.23s
> Friday 03 September 2021  11:03:49 +0200 (0:00:00.027)       0:02:37.282 ****** 
> =============================================================================== 
> prerequisites - docker - dnf-plugins-core ----------------------------------------------------------------------------------------------- 74.15s
> prerequisites - docker - packages ------------------------------------------------------------------------------------------------------- 41.56s
> prerequisites - packages - git jq java -------------------------------------------------------------------------------------------------- 36.22s
> procedure - clone the repository --------------------------------------------------------------------------------------------------------- 2.24s
> prerequisites - docker - service --------------------------------------------------------------------------------------------------------- 2.12s
> prerequisites - docker - repository ------------------------------------------------------------------------------------------------------ 0.91s
> procedure - interactive steps ------------------------------------------------------------------------------------------------------------ 0.03s
> Playbook run took 0 days, 0 hours, 2 minutes, 37 seconds
> Warning: Permanently added '192.168.123.5' (ED25519) to the list of known hosts.
> Last login: Fri Sep  3 09:03:47 2021 from 192.168.123.1
> [ffloreth@fedora-34 ~]$ cd jetbrains-editor-images
> [ffloreth@fedora-34 jetbrains-editor-images]$ ./projector.sh build
> [info] Select the IDE package to build (default is 'IntelliJ IDEA Community'):
>        1) IntelliJ IDEA Community
>        2) PyCharm Community
> #? 1
> [info] Select the IDE package version to build (default is '2020.3.3'):
>        1) 2020.3.3
>        2) 2020.3.2
>        3) 2020.3.1
> #? 1
> [info] Read the container name 'che-idea:2020.3.3'
> [info] Read the URL for IDE packaging 'https://download-cdn.jetbrains.com/idea/ideaIC-2020.3.3.tar.gz'
> [info] Check for installed Java
> [info] Read JAVA_HOME env: 
> [info] Read PATH env: /home/ffloreth/.local/bin:/home/ffloreth/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin
> [info] Found 'java' executable in PATH
> [info] Java version: 16.0.2
> --2021-09-03 09:07:51--  https://download-cdn.jetbrains.com/idea/ideaIC-2020.3.3.tar.gz
> Resolving download-cdn.jetbrains.com (download-cdn.jetbrains.com)... 13.225.19.37, 13.225.19.94, 13.225.19.43, ...
> Connecting to download-cdn.jetbrains.com (download-cdn.jetbrains.com)|13.225.19.37|:443... connected.
> HTTP request sent, awaiting response... 200 OK
> Length: 734643423 (701M) [binary/octet-stream]
> Saving to: ‘ideaIC-2020.3.3.tar.gz’
> 
> ideaIC-2020.3.3.tar.gz               100%[===================================================================>] 700.61M  8.97MB/s    in 67s     
> 
> 2021-09-03 09:08:59 (10.4 MB/s) - ‘ideaIC-2020.3.3.tar.gz’ saved [734643423/734643423]
> 
> [info] Check Projector Client source directory
> [warning] Projector Client source directory '/home/ffloreth/jetbrains-editor-images/build/projector-client' doesn't exist
> [info] Cloning Projector Client sources to '/home/ffloreth/jetbrains-editor-images/build/projector-client'
> [info] Checkout Projector Client to SHA1 'd80f9952cc520102bb2de6f1aefaad6aadb49138'
> [info] HEAD is now at d80f995 PRJ-541 PRJ-542 Show CloseEvent details on disconnection
> [info] Applying patches for Projector Client
>        Patching Projector Client with '/home/ffloreth/jetbrains-editor-images/patches/projector-client/PRJ-452.patch'
> [info] Check Projector Server source directory
> [warning] Projector Server source directory '/home/ffloreth/jetbrains-editor-images/build/projector-server' doesn't exist
> [info] Cloning Projector Server sources to '/home/ffloreth/jetbrains-editor-images/build/projector-server'
> [info] Checkout Projector Client to SHA1 'e8ecfda9ee04b59e8b4276215e22a3cbff212083'
> [info] HEAD is now at e8ecfda New release
> [info] Build Projector on localhost
> The message received from the daemon indicates that the daemon has disappeared.
> Build request sent: Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server}
> Attempting to read last messages from the daemon log...
> Daemon pid: 11250
>   log file: /home/ffloreth/.gradle/daemon/6.4.1/daemon-11250.out.log
> ----- Last  20 lines from daemon log file - daemon-11250.out.log -----
> 2021-09-03T09:09:20.700+0000 [INFO] [org.gradle.launcher.daemon.server.DefaultIncomingConnectionHandler] Received command: Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server}.
> 2021-09-03T09:09:20.700+0000 [DEBUG] [org.gradle.launcher.daemon.server.DefaultIncomingConnectionHandler] Starting executing command: Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server} with connection: socket connection from /127.0.0.1:44471 to /127.0.0.1:57178.
> 2021-09-03T09:09:20.703+0000 [ERROR] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] Command execution: started DaemonCommandExecution[command = Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server}, connection = DefaultDaemonConnection: socket connection from /127.0.0.1:44471 to /127.0.0.1:57178] after 0.0 minutes of idle
> 2021-09-03T09:09:20.703+0000 [INFO] [org.gradle.launcher.daemon.server.DaemonRegistryUpdater] Marking the daemon as busy, address: [980bd925-1bbb-444d-8a01-e5f3f87f7f35 port:44471, addresses:[localhost/127.0.0.1]]
> 2021-09-03T09:09:20.703+0000 [DEBUG] [org.gradle.launcher.daemon.registry.PersistentDaemonRegistry] Marking busy by address: [980bd925-1bbb-444d-8a01-e5f3f87f7f35 port:44471, addresses:[localhost/127.0.0.1]]
> 2021-09-03T09:09:20.706+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire exclusive lock on daemon addresses registry.
> 2021-09-03T09:09:20.706+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
> 2021-09-03T09:09:20.708+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
> 2021-09-03T09:09:20.709+0000 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] resetting idle timer
> 2021-09-03T09:09:20.709+0000 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] daemon is running. Sleeping until state changes.
> 2021-09-03T09:09:20.711+0000 [INFO] [org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy] Daemon is about to start building Build{id=6d106e88-4dfa-4ed7-99b7-2082a4704269, currentDir=/home/ffloreth/jetbrains-editor-images/build/projector-server}. Dispatching build started information...
> 2021-09-03T09:09:20.711+0000 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 21: dispatching org.gradle.launcher.daemon.protocol.BuildStarted@582ccbc9
> 2021-09-03T09:09:20.714+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment] Configuring env variables: {PATH=/home/ffloreth/.local/bin:/home/ffloreth/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin, LC_MEASUREMENT=C.UTF-8, LC_MONETARY=C.UTF-8, HISTCONTROL=ignoredups, XMODIFIERS=@im=ibus, LC_TIME=C.UTF-8, HISTSIZE=1000, BASH_FUNC_which%%=() {  ( alias;
>  eval ${_declare} ) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot "$@"
> }, TERM=xterm-256color, MOTD_SHOWN=pam, DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus, LANG=en_US.UTF-8, XDG_SESSION_ID=3, XDG_SESSION_TYPE=tty, SELINUX_LEVEL_REQUESTED=, SELINUX_ROLE_REQUESTED=, MAIL=/var/spool/mail/ffloreth, LC_PAPER=C.UTF-8, LOGNAME=ffloreth, PWD=/home/ffloreth/jetbrains-editor-images/build/projector-server, XDG_SESSION_CLASS=user, LESSOPEN=||/usr/bin/lesspipe.sh %s, SHELL=/bin/bash, SELINUX_USE_CURRENT_RANGE=, SSH_TTY=/dev/pts/0, SSH_CLIENT=192.168.123.1 41862 22, OLDPWD=/home/ffloreth/jetbrains-editor-images/build/projector-server, USER=ffloreth, LC_NUMERIC=C.UTF-8, SSH_CONNECTION=192.168.123.1 41862 192.168.123.5 22, HOSTNAME=fedora-34, XDG_RUNTIME_DIR=/run/user/1000, LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=01;37;41:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=01;36:*.au=01;36:*.flac=01;36:*.m4a=01;36:*.mid=01;36:*.midi=01;36:*.mka=01;36:*.mp3=01;36:*.mpc=01;36:*.ogg=01;36:*.ra=01;36:*.wav=01;36:*.oga=01;36:*.opus=01;36:*.spx=01;36:*.xspf=01;36:, SHLVL=2, HOME=/home/ffloreth}
> 2021-09-03T09:09:20.730+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.LogToClient] About to start relaying all logs to the client via the connection.
> 2021-09-03T09:09:20.730+0000 [INFO] [org.gradle.launcher.daemon.server.exec.LogToClient] The client will now receive all logging from the daemon (pid: 11250). The daemon log file: /home/ffloreth/.gradle/daemon/6.4.1/daemon-11250.out.log
> 2021-09-03T09:09:20.733+0000 [INFO] [org.gradle.launcher.daemon.server.exec.LogAndCheckHealth] Starting build in new daemon [memory: 519.0 MB]
> 2021-09-03T09:09:20.744+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] The daemon has started executing the build.
> 2021-09-03T09:09:20.747+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] Executing build with daemon context: DefaultDaemonContext[uid=89502f5f-bcdb-432e-9a05-6f26dcbe36ab,javaHome=/usr/lib/jvm/java-16-openjdk-16.0.2.0.7-1.rolling.fc34.x86_64,daemonRegistryDir=/home/ffloreth/.gradle/daemon,pid=11250,idleTimeout=10800000,priority=NORMAL,daemonOpts=--add-opens,java.base/java.util=ALL-UNNAMED,--add-opens,java.base/java.lang=ALL-UNNAMED,--add-opens,java.base/java.lang.invoke=ALL-UNNAMED,--add-opens,java.prefs/java.util.prefs=ALL-UNNAMED,-XX:MaxMetaspaceSize=256m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=UTF-8,-Duser.country=US,-Duser.language=en,-Duser.variant]
> ----- End of the daemon log -----
> 
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> find: ‘projector-server/build/distributions’: No such file or directory
> [info] Build 'che-idea:2020.3.3'
> Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/build?buildargs=%7B%7D&buildid=36b96f3d31493cf73e0fd8a119343c73961690ef101e540bf58ae5e229f68365&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=default&remote=client-session&rm=1&session=ommuhhgz7j2d2iq9lpxa7mo8k&shmsize=0&t=che-idea%3A2020.3.3&target=&ulimits=null&version=2": dial unix /var/run/docker.sock: connect: permission denied
> [warning] Container build failed
> [ffloreth@fedora-34 jetbrains-editor-images]$ free
>                total        used        free      shared  buff/cache   available
> Mem:          734072      161372      445768        1356      126932      462012
> Swap:              0           0           0
> ```

Will take a look today, there's something unusual.

Reply:
It looks much better using `java-11-openjdk` instead of the latest jdk wich is version 16. Now at  

```
[info] Build Projector on localhost
<-------------> 0% INITIALIZING [15m 35s]
> Evaluating settings > Compiling settings file '/home/ffloreth/jetbrains-editor-images/build/projector-client/settings.gradle.kts' into local c
```